### ISSUE TITLE
Compilation does not work on aarch64

### ISSUE THREAD

**Issue by fingon** [timescale/promscale_extension#30]
```
   Compiling pgx v0.1.19 (https://github.com/JLockerman/pgx.git?branch=guardless#c8248249)             
error[E0308]: mismatched types                                                                         
  --> /home/mstenber/.cargo/git/checkouts/pgx-24f98968818058e4/c824824/pgx/src/enum_helper.rs:33:43    
   |                                                                                                   
33 |         unsafe { std::ffi::CStr::from_ptr(en.enumlabel.data.as_ptr() as *const i8) }              
   |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u8`, found `i8`                                                                                              
   |                                                                                                   
   = note: expected raw pointer `*const u8`                                                            
              found raw pointer `*const i8`                                                            
                                                                                                       
error[E0308]: mismatched types                                                                         
  --> /home/mstenber/.cargo/git/checkouts/pgx-24f98968818058e4/c824824/pgx/src/bgworkers.rs:70:17      
   |                                                                                                   
70 |                 &(*pg_sys::MyBgworkerEntry).bgw_name,                                             
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i8`, found `u8`                    
   |                                                                                                   
   = note: expected reference `&[i8; 96]`                                                                                                                                                                     
              found reference `&[u8; 96]`                                                              
                                                                                                       
```

**Comment by JamesGuthrie** [comment]
@fingon we just released version 0.3.0 of the extension, could you try that out and see if it compiles?

(Bear in mind that we don't have a compatible release of `promscale` yet, that's still in progress)

**Comment by fingon** [comment]
It seems to be fixed in 0.3.0 (pg12, pg13 compile); there seems to be some other issue with cargo pgx and pg14, but I filed a bug there instead about it - https://github.com/zombodb/pgx/issues/308

**Comment by JamesGuthrie** [comment]
Thanks for the feedback 👍 

FYI we're tracking the next promscale release (which will "officially" support version 0.3.0 of the extension) here: https://github.com/timescale/promscale/issues/890.

If you compile and install version 0.3.0 of the extension and use it with current promscale it will probably work, but that's not really recommended.
